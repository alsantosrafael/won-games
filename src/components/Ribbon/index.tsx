import * as S from './styles';
import { RibbonColors, RibbonSizes } from './styles';

export type RibbonProps = {
  children: React.ReactNode
  color?: RibbonColors
  size?: RibbonSizes
}
const Ribbon = ({ children, color = 'primary', size = 'normal' }: RibbonProps) => (
  <S.Wrapper
    color={color}
    size={size}
  >
		{children}
  </S.Wrapper>
);
export default Ribbon;
