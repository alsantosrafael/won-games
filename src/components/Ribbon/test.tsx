import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Ribbon from '.';

describe('<Ribbon />', () => {
  it('should render the heading', () => {
    renderWithTheme(<Ribbon>Best Seller</Ribbon>)
    expect(screen.getByText(/best seller/i),).toBeInTheDocument();
  });

  it('should render component with primary color', () => {
    const { container } = renderWithTheme(<Ribbon>Best Seller</Ribbon>)
    expect(screen.getByText(/best seller/i)).toHaveStyle({
        backgroundColor: '#F231A5'
      });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component with primary color', () => {
    const { container } = renderWithTheme(<Ribbon color='secondary'>Best Seller</Ribbon>)
    expect(screen.getByText(/best seller/i)).toHaveStyle({
        backgroundColor: '#3CD3C1'
      });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component with normal size as default', () => {
    const { container } = renderWithTheme(<Ribbon color='secondary'>Best Seller</Ribbon>)
    expect(screen.getByText(/best seller/i)).toHaveStyle({
      fontSize: '1.4rem',
      height: '3.6rem'
    })
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component with small size when passed props', () => {
    const { container } = renderWithTheme(<Ribbon size='small'>Best Seller</Ribbon>)
    expect(screen.getByText(/best seller/i)).toHaveStyle({
      fontSize: '1.2rem',
      height: '2.6rem'
    })
    expect(container.firstChild).toMatchSnapshot();
  });
});


