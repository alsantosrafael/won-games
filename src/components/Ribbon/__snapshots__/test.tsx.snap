// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> should render component with normal size as default 1`] = `
.c0 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #3CD3C1;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 0 2.4rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #1f8a7d;
  border-top-color: #1f8a7d;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<main
  class="c0"
  color="secondary"
>
  Best Seller
</main>
`;

exports[`<Ribbon /> should render component with primary color 1`] = `
.c0 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #F231A5;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 0 2.4rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #b20b6f;
  border-top-color: #b20b6f;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<main
  class="c0"
  color="primary"
>
  Best Seller
</main>
`;

exports[`<Ribbon /> should render component with primary color 2`] = `
.c0 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #3CD3C1;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 0 2.4rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #1f8a7d;
  border-top-color: #1f8a7d;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

<main
  class="c0"
  color="secondary"
>
  Best Seller
</main>
`;

exports[`<Ribbon /> should render component with small size when passed props 1`] = `
.c0 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #F231A5;
  font-size: 1.2rem;
  height: 2.6rem;
  padding: 0 1.6rem;
  right: -1.5rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c0::before {
  border-left-color: #b20b6f;
  border-top-color: #b20b6f;
}

.c0::before {
  top: 2.6rem;
  border-top-width: 0.7rem;
  border-right-width: 1.5rem;
}

<main
  class="c0"
  color="primary"
>
  Best Seller
</main>
`;
